src/NServiceBus.Core/InitializableEndpoint.cs

```diff
@@ -126,16 +126,12 @@ void RunUserRegistrations(List<Action<IConfigureComponents>> registrations)
 
         void RegisterContainerAdapter(IContainer containerToAdapt)
         {
-            var b = new CommonObjectBuilder
-            {
-                Container = containerToAdapt
-            };
+            var b = new CommonObjectBuilder(containerToAdapt);
```

Comment:
I realized that I don't like this temporal coupling between `RegisterContainerAdapter` and `RegisterRegistrations`. The `RegisterContainerAdapter` is called in the ctor and sets the `builder` and the `container` field. Then the `RegisterRegistrations` is called which uses that field. I Propose we change the design to something like

* `RegisterContainerAdapter` creates the CommonObjectBuilder, registers it as `IBuilder` and returns it
* Constructor sets a field of type `CommonObjectBuilder`
* Ctor calls `RegisterRegistrations` and passes in the common object builder